src/other/tables.jl

```diff
@@ -7,16 +7,24 @@ Tables.rows(df::AbstractDataFrame) = eachrow(df)
 Tables.schema(df::AbstractDataFrame) = Tables.Schema(names(df), eltype.(eachcol(df)))
 Tables.materializer(df::AbstractDataFrame) = DataFrame
 
+Tables.getcolumn(df::AbstractDataFrame, i::Int) = df[!, i]
+Tables.getcolumn(df::AbstractDataFrame, nm::Symbol) = df[!, nm]
+Tables.columnnames(df::AbstractDataFrame) = names(df)
+
+Tables.getcolumn(dfr::DataFrameRow, i::Int) = dfr[i]
+Tables.getcolumn(dfr::DataFrameRow, nm::Symbol) = dfr[nm]
+Tables.columnnames(dfr::DataFrameRow) = names(dfr)
+
 getvector(x::AbstractVector) = x
-getvector(x) = collect(x)
+getvector(x) = [x[i] for i = 1:length(x)]
```

Comment:
what could the type of `x` in practice? (I guess it is an iterable that is not an `AbstractVector` but has length and is indexable; then what is left here except `Tuple` and `NamedTuple`). I understand `collect` allowed iterables that were not indexable - right?

Reply:
@quinnj - can you please comment on this - so that I have a better understanding of the intention of this method. Thank you!

Reply:
For Tables.jl 1.0, a column must be an indexable collection with known length. So `[x[i] for i = 1:length(x)]` is the most correct in terms of what the interface requires.

Reply:
Though `collect` could be faster for some types, right? I just saw you mentioned on Slack that it was faster to iterate over JSON arrays than to call `getindex` on each entry. :-p